### DISCUSSION TITLE
Improve Combobox component

### DISCUSSION THREAD

**Discussion by MadadiSaeed** [chakra-ui/zag#932] [Ideas]
Hi. For the last two days I am trying to figure out a way to build a combobox component similar to this mantine component that has a chip mode and you can search for options.
![Screenshot from 2023-10-09 17-33-02](https://github.com/chakra-ui/zag/assets/18693428/19bbc67e-4e8e-417b-a670-a598353d9ce2)
But unfortunately I wasn't successful. Until now I was just fighting Combobox's input element to let me just add my custom text value (better control). Whenever I select or unfocus from the input it reset input's value based on selected options plus there is no hidden inpu to keep selected items when it is used inside a form. But I don't want selected options to be shown as input's value. Here is my suggestion to have more flexible combobox component:
1. Add a hidden input element that keeps track of selected options or Add hidden html `select` element similar to hidden select element inside `Select` Component. 
2. Add a normal input element that has all the event listeners of the current input but gives us controling the value of the input elements. 
3. Of course these two input elements should have accessible features.

**Comment by segunadebayo** [comment]
This is largely a documentation issue. We already gave the tags input and combo box, the main task is to compose them together to re-create this UX you've described.

We'll consider adding this to the docs.

**Reply by kazimirchuk** [reply]
@segunadebayo was there documentation added on this front? I couldn't find one. I can see there's a property `allowCustomValue` however when I use it and let's say press enter after typing the custom value, the value itself isn't injected into the collection and isn't properly "selected" if it makes sense. When I submit the form the actual value keeps being empty. I'm sure it's possible to achieve that I just think I'm missing something real simple
